virt_lightning/shell.py

```diff
@@ -335,6 +337,48 @@ def storage_dir(configuration, **kwargs):
     print(hv.get_storage_dir())  # noqa: T001
 
 
+def fetch(configuration, **kwargs):
+    conn = libvirt.open(configuration.libvirt_uri)
+    hv = vl.LibvirtHypervisor(conn)
+    hv.init_storage_pool(configuration.storage_pool)
+    storage_dir = hv.get_storage_dir()
+    conn = http.client.HTTPSConnection("virt-lightning.org")
+    conn.request("GET", "/images/{distro}/{distro}.qcow2".format(**kwargs))
+    r = conn.getresponse()
+    if r.status == 404:
+        print("Distro {distro} not found!".format(**kwargs))  # noqa: T001
+        print(  # noqa: T001
+            "Visit https://virt-lightning.org/images/ to get an up to date list."
+        )
+        sys.exit(1)
+    lenght = int(r.headers["Content-Length"])
+    MB = 1024 * 1000
+    chunk_size = MB
+    target_file = pathlib.PosixPath(
+        "{storage_dir}/upstream/{distro}.qcow2".format(
+            storage_dir=storage_dir, **kwargs
+        )
+    )
+    temp_file = target_file.with_suffix(".temp")
+    if target_file.exists():
+        print(  # noqa: T001
+            "File already exists: {target_file}".format(target_file=target_file)
+        )
+        sys.exit(0)
+    with temp_file.open("wb") as fd:
+        while fd.tell() < lenght:
+            chunk = r.read(chunk_size)
+            fd.write(chunk)
+            percent = (fd.tell() * 100) / lenght
+            line = "[{percent:06.2f}%]  {done:6}MB/{full}MB\r".format(
```

Comment:
This code might change to use quantiphy library.

Reply:
I would prefer to avoid any third-party dep for now.